barman/cloud_providers/__init__.py

```diff
@@ -69,6 +67,25 @@ def _make_s3_cloud_interface(config, cloud_interface_kwargs):
     return S3CloudInterface(**cloud_interface_kwargs)
 
 
+def _get_azure_credential(credential_type):
```

Comment:
This will cause `azure.identity` to be imported even if the credential type is `None` which will break barman cloud with Azure Blob Storage in deployments where `azure-identity` is not available - that would be a change in behaviour so we should avoid it.